pacman/model/resources/variable_sdram.py

```diff
@@ -0,0 +1,76 @@
+from .abstract_sdram import AbstractSDRAM
+from .constant_sdram import ConstantSDRAM
+from pacman.exceptions import PacmanConfigurationException
+
+
+class VariableSDRAM(AbstractSDRAM):
+    """ Represents an amount of SDRAM used on a chip in the machine.
+
+    This is where the usage increase as the run time increases
+    """
+
+    __slots__ = [
+        # The amount of SDRAM in bytes used no matter what
+        "_fixed_sdram"
+        # The amount of extra SDRAm used for each timestep
+        "_per_timestep_sdram"
+    ]
+
+    def __init__(
+            self, fixed_sdram, per_timestep_sdram):
+        """
+        :param sdram: The amount of SDRAM in bytes
+        :type sdram: int
+        :raise None: No known exceptions are raised
+        """
+        self._fixed_sdram = fixed_sdram
+        self._per_timestep_sdram = per_timestep_sdram
+
+    def get_total_sdram(self, n_timesteps):
+        if n_timesteps:
```

Comment:
It n_timespetps is zero then 1. Something has gone very wrong to even get in that state.  2. The code raises and exception anyway.  So not changing

Reply:
Apparently n_timesteps=0 is valid (and currently works)

Reply:
There is a question as to whether *this* code should ever receive 0; would this imply that no timesteps can be run at all due to the memory? I guess if the user specified the minimum run timesteps to be 0 and then the code filled the SDRAM, this would still be an error state but the error would be confusing at best

Reply:
Even if run(0) is called the cost models are still computed at minimum plan timestep which would be dangerous so extremely unlikely.  Not impossible and technically correct so fixed  to keep the fiddler happy.